src/combat.cpp

```diff
@@ -922,19 +924,19 @@ void Combat::doAreaCombat(Creature* caster, const Position& position, const Area
 
 	postCombatEffects(caster, position, params);
 
-	std::vector<Creature*> toDamageCreatures;
+	std::vector<std::shared_ptr<Creature>> toDamageCreatures;
 	toDamageCreatures.reserve(100);
 
-	for (Tile* tile : tiles) {
+	for (const auto& tile : tiles) {
 		if (canDoCombat(caster, tile, params.aggressive) != RETURNVALUE_NOERROR) {
 			continue;
 		}
 
 		combatTileEffects(spectators, caster, tile, params);
 
 		if (CreatureVector* creatures = tile->getCreatures()) {
-			const Creature* topCreature = tile->getTopCreature();
-			for (Creature* creature : *creatures) {
+			auto topCreature = tile->getTopCreature();
```

Comment:
Could this be `const auto`
```suggestion
			const auto topCreature = tile->getTopCreature();
```